### ISSUE TITLE
unused-lifetimes lint false-positives with async functions

### ISSUE THREAD

**Issue by Seeker14491** [rust-lang/rust#61115]
Async functions that take references seem to always trigger the `unused-lifetimes` lint (which is allow-by-default). Example:

```rust
#![feature(async_await)]

#![warn(unused_lifetimes)]

pub async fn print(s: &str) {
    println!("{}", s);
}
```

```
warning: lifetime parameter `'_` never used
 --> src/lib.rs:5:23
  |
5 | pub async fn print(s: &str) {
  |                       ^
  |
```

**Comment by Seeker14491** [comment]
I found [this clippy issue](https://github.com/rust-lang/rust-clippy/issues/3988) which describes basically the same problem.

**Comment by nikomatsakis** [comment]
Seeing as this lint is not enabled by default, marking as deferred, though this would be good to fix and probably not too hard.

**Comment by gilescope** [comment]
Looks the same issue as #64493. 

A little inelegant but if
```
    !"5 | pub async fn print(s: &str) {".contains("'_")
```
we can suppress the lint I think...
